agent/connect/ca/provider_aws.go

```diff
@@ -370,7 +370,7 @@ func (a *AWSProvider) loadCACerts() error {
 		if output.CertificateChain == nil {
 			return fmt.Errorf("Subordinate CA %s returned no chain", a.arn)
 		}
-		a.rootPEM = *output.CertificateChain
```

Comment:
This looks unrelated, is this necessary for the trailing newline fix, or is this a separate fix, or accidental?